Bottom: ee2a8df4f5cae53c8b3835a341722597ab432d14
Top:    9e6f6d20209609d4abd22fe7096dfdb2624f579d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 02:37:57 +0000

configure.ac: Save and restore CPPFLAGS either side of test for X11 stuff

We add X_CFLAGS and X_LIBS to the main flag list later, so don't leave
an extra copy lying around in the CPPFLAGS variable.


---

diff --git a/configure.ac b/configure.ac
index 5b202b1..eb6d238 100644
--- a/configure.ac
+++ b/configure.ac
@@ -705,6 +705,7 @@ for e in $HIDLIST; do
     case $e in
       lesstif )
 	AC_PATH_XTRA
+	save_CPPFLAGS="$CPPFLAGS"
 	CPPFLAGS="$X_CFLAGS"
 	AC_CHECK_LIB(X11, XOpenDisplay, , , $X_LIBS)
 	AC_CHECK_LIB(ICE, main, , , $X_LIBS)
@@ -713,7 +714,8 @@ for e in $HIDLIST; do
 	AC_CHECK_LIB(Xt, XtOpenDisplay, , , $X_LIBS)
 	AC_CHECK_LIB(Xmu, main, , , $X_LIBS)
 	AC_CHECK_LIB(Xpm, main, , , $X_LIBS)
-        AC_CHECK_LIB(Xm, XmCreateMainWindow, , , $X_LIBS)
+	AC_CHECK_LIB(Xm, XmCreateMainWindow, , , $X_LIBS)
+	CPPFLAGS="$save_CPPFLAGS"
 	case $ac_cv_lib_Xm_XmCreateMainWindow in
 	  no )
 	    AC_MSG_ERROR([You don't seem to have the Lesstif development environment installed.])
